meson.build

```diff
@@ -0,0 +1,54 @@
+# This file is adapted from https://github.com/scipy/scipy/blob/main/meson.build
+project(
+    'pandas',
+    'c', 'cpp', 'cython',
+    version: '2.0.0.dev0',
+    license: 'BSD-3',
+    meson_version: '>=0.64',
+    default_options: [
+        # TODO: investigate, does meson try to compile against debug Python
+        # when buildtype = debug, this seems to be causing problems on CI
+        # where provided Python is not compiled in debug mode
+        'buildtype=release',
+        # TODO: turn on werror when ready
+        #'werror=true',
+        # TODO: We are using POSIX functions(strdup in ujson), so we can't compile
+        # with strict C99 :(
+        #'c_std=c99'
+    ]
+)
+
+add_global_arguments('-DNPY_NO_DEPRECATED_API=0', language : 'c')
+add_global_arguments('-DNPY_NO_DEPRECATED_API=0', language : 'cpp')
```

Comment:
These should probably be `add_project_arguments` (and below as well). The difference is subtle, it's only going to matter if pandas is built as part of a larger project or if it gains meson subprojects. Then `global` propagates outside of the pandas build itself.